mlir/lib/Dialect/MIOpen/Transforms/GridwiseGemmToBlockwise.cpp

```diff
@@ -84,6 +84,21 @@ void computeLoadStoreTypeInfo(OpBuilder &b, ArrayRef<int64_t> sliceLengths,
   loadType = vectorTypeOrSelf(elementType, loadLength);
 }
 
+Type obtainAccumulatorType(OpBuilder &b, Type &elementType, Type &destType) {
+  // Determine the type used on VGPR to act as accumulator.
+  // f32: f32.
+  // f16, bf16: f32 to prevent overflow from happening.
+  // i16 : i16.
+  // i8: i32, since we have an i32 output
+  Type accumulatorType = destType;
+  if (elementType == b.getF16Type() || elementType == b.getBF16Type()) {
```

Comment:
Minor comments I realized, we have `elementType.isF15()`  and such. We could also, in principle, not pass in the builder and use stuff like the `get()` methods on the types we want to construct, passing in the context from the elementType, but idk about if that's the right call here.

Reply:
I like the idea of check type from themselves instead of creating a useless temporary and have them immediately destroyed, good point!

On the second one, all type construction require a context, which is probably why those have been abstracted away in `Builders.h` as a convenient wrapper. I'll skip this one and continue to pass in the builder.